File: frontend/src/app/models/event.ts
export interface Event {
  id?: number;
  title: string;
  start: Date;
  end: Date;
  all_day: boolean;
  team_id: number;
}

export interface CalendarEvent {
  title: string;
  start: Date | string;
  end: Date | string;
  allDay: boolean;
  id: number | undefined;
}

File: backend/lib/time_manager/events/event.ex
defmodule TimeManager.Events.Event do
  use Ecto.Schema
  import Ecto.Changeset

  @derive {Jason.Encoder, only: [:id, :start, :end, :title, :all_day, :team_id]}
  schema "events" do
    field :start, :utc_datetime
    field :title, :string
    field :end, :utc_datetime
    field :all_day, :boolean, default: false
    field :team_id, :id

    timestamps(type: :utc_datetime)
  end

  @doc false
  def changeset(event, attrs) do
    event
    |> cast(attrs, [:title, :start, :end, :all_day, :team_id])
    |> validate_required([:title, :start, :team_id])
  end
end
